internal/daemon/api_shutdown.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"net/http"
+
+	"github.com/canonical/pebble/internal/logger"
+)
+
+// v1PostShutdown is a handler for initiating a Pebble shutdown via the daemon.
+func v1PostShutdown(c *Command, r *http.Request, _ *userState) Response {
+	logger.Noticef("received shut down api request, proceeding to shut down daemon")
```

Comment:
Nit: we've tried to be careful to keep log message formatting consistent, starting log messages with a capital letter (unlike errors), and using proper capitalization like "API".

However, I don't think we actually need this log, as all API requests are logged. So even without this we get something like this (the `POST /v1/shutdown` line is the one I'm referring to):

```
2023-02-06T23:03:57.176Z [pebble] POST /v1/shutdown 50.11µs 200
2023-02-06T23:03:57.176Z [pebble] Server exiting!
2023-02-06T23:03:57.201Z [pebble] Stopping all running services.
2023-02-06T23:03:57.231Z [pebble] Service "srv1" stopped
```